agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/AiConfigCustomizer.java

```diff
@@ -150,10 +169,20 @@ private static void enableInstrumentations(
     properties.put("otel.instrumentation.kotlinx-coroutines.enabled", "true");
     properties.put("otel.instrumentation.liberty.enabled", "true");
     properties.put("otel.instrumentation.liberty-dispatcher.enabled", "true");
-    properties.put("otel.instrumentation.log4j-appender.enabled", "true");
-    if (otelConfig.getBoolean("otel.instrumentation.logback-appender.enabled", true)) {
-      properties.put("otel.instrumentation.logback-appender.enabled", "true");
+    if (config.instrumentation.log4jAppender.enabled) {
+      properties.put("otel.instrumentation.log4j-appender.enabled", "true");
+    } else {
+      properties.put("otel.instrumentation.log4j-appender.enabled", "false");
+    }
+    if (config.instrumentation.logbackAppender.enabled) {
+      if (otelConfig.getBoolean("otel.instrumentation.logback-appender.enabled", true)) {
```

Comment:
Could you explain the need for `otelConfig.getBoolean("otel.instrumentation.logback-appender.enabled", true)`?

Reply:
i don't know the default value upstream. It's better to be explicit.